client/my-sites/email/test/paths.ts

```diff
@@ -0,0 +1,237 @@
+			`/email//titan/control-panel/${ siteName }`
+		);
+		expect( emailManagementTitanControlPanelRedirect( null, '' ) ).toEqual(
+			`/email//titan/control-panel/null`
+		);
+	} );
+
+	it( 'emailManagement', () => {
+		expect( emailManagement( siteName, domainName ) ).toEqual(
+			`/email/${ domainName }/manage/${ siteName }`
+		);
+		expect( emailManagement( ':site', ':domain', emailManagementAllSitesPrefix ) ).toEqual(
+			'/email/all/:domain/manage/:site'
+		);
+		expect( emailManagement( ':site', ':domain' ) ).toEqual( '/email/:domain/manage/:site' );
+		expect( emailManagement( ':site' ) ).toEqual( '/email/:site' );
+		expect( emailManagement( siteName, null ) ).toEqual( `/email/${ siteName }` );
+		expect( emailManagement( null, null ) ).toEqual( `/email` );
+	} );
+
+	it( 'emailManagementForwarding', () => {
+		expect( emailManagementForwarding( siteName, domainName ) ).toEqual(
+			`/email/${ domainName }/forwarding/${ siteName }`
+		);
+		expect(
+			emailManagementForwarding( ':site', ':domain', emailManagementAllSitesPrefix )
+		).toEqual( '/email/all/:domain/forwarding/:site' );
+		expect( emailManagementForwarding( ':site', ':domain' ) ).toEqual(
+			'/email/:domain/forwarding/:site'
+		);
+		expect( emailManagementForwarding( siteName, '' ) ).toEqual(
+			`/email//forwarding/${ siteName }`
+		);
+		expect( emailManagementForwarding( null, '' ) ).toEqual( `/email//forwarding/null` );
+	} );
+
+	it( 'emailManagementPurchaseNewEmailAccount', () => {
+		expect( emailManagementPurchaseNewEmailAccount( siteName, domainName ) ).toEqual(
+			`/email/${ domainName }/purchase/${ siteName }`
+		);
+		expect(
+			emailManagementPurchaseNewEmailAccount( ':site', ':domain', emailManagementAllSitesPrefix )
+		).toEqual( '/email/all/:domain/purchase/:site' );
+		expect( emailManagementPurchaseNewEmailAccount( ':site', ':domain' ) ).toEqual(
+			'/email/:domain/purchase/:site'
+		);
+		expect( emailManagementPurchaseNewEmailAccount( siteName, '' ) ).toEqual(
+			`/email//purchase/${ siteName }`
+		);
+		expect( emailManagementPurchaseNewEmailAccount( null, '' ) ).toEqual( `/email//purchase/null` );
+	} );
+
+	it( 'emailManagementInDepthComparison', () => {
+		expect( emailManagementInDepthComparison( siteName, domainName ) ).toEqual(
+			`/email/${ domainName }/compare/${ siteName }`
+		);
+		expect( emailManagementInDepthComparison( ':site', ':domain' ) ).toEqual(
+			'/email/:domain/compare/:site'
+		);
+		expect( emailManagementInDepthComparison( siteName, '' ) ).toEqual(
+			`/email//compare/${ siteName }`
+		);
+		expect( emailManagementInDepthComparison( null, '' ) ).toEqual( `/email//compare/null` );
+	} );
+
+	it( 'emailManagementMailboxes', () => {
+		expect( emailManagementMailboxes( siteName ) ).toEqual( `/mailboxes/${ siteName }` );
+		expect( emailManagementMailboxes( ':site' ) ).toEqual( '/mailboxes/:site' );
+		expect( emailManagementMailboxes() ).toEqual( '/mailboxes' );
+	} );
+
+	it.each( [
```

Comment:
In general cool tests, with tiny IMO exception about this one. More details here https://github.com/Automattic/wp-calypso/pull/86096#discussion_r1445957298